### DISCUSSION TITLE
using assign_priority_on_intersections does not adjust priorities automatically

### DISCUSSION THREAD

**Discussion by aknsj** [ansys/pyaedt#4378] [Q&A]
Hi!I am trying to use the following code to adjust priorities automatically. It returns True, but the priority in IcepakDesign is not adjusted and clicking on "Validate" still reports intersect errors.
ipk.assign_priority_on_intersections()
PyAEDT INFO: Project MX11 Saved correctly
PyAEDT INFO: 0 Intersections have been found. Applying Priorities
Out[53]: True
![image](https://github.com/ansys/pyaedt/assets/163245306/afcc21d6-2e88-4382-a652-4ab9d2a58141)
![image](https://github.com/ansys/pyaedt/assets/163245306/32de52d0-0635-40b6-a512-d6688842d575)
The API guide for assign_priority_on_intersections is as follows, saying that it will automatically adjust the priority if there are intersections.
![image](https://github.com/ansys/pyaedt/assets/163245306/82da2119-e2fb-40cd-bcd3-22f50fe38b4e)
Thank you in advance for your help and guidance!

**Comment by Samuelopez-ansys** [comment]
@aknsj The method is checking the intersections with components using a name convention. By default is "COMP_", you can debug, and see how it works.
https://github.com/ansys/pyaedt/blob/main/pyaedt/icepak.py#L1099

Maybe you just need to change the default prefix.

@lorenzovecchietti

**Reply by aknsj** [reply]
Thank you for your reply.
First, I found that running the following code calls the add_priority function in MeshIcepak.py.

        ipk.assign_priority_on_intersections(component_prefix="plastic")

Then, in the add_priority function, the object named "plastic" reports the error 'NoneType' object has no attribute 'object_type' when running the following code.

        prio = [
                "NAME:PriorityListParameters",
                "EntityType:=",
                "Object",
                "EntityList:=",
                objects,
                "PriorityNumber:=",
                i,
                "PriorityListType:=",
                ["2D", "3D"][int(self._app.modeler[new_obj_list[0]].is3d)],]
            ]

Then I read the object_type function  in object3d.py and found the following code which is the code to determine the type of the object.

        if self._m_name in list(self._oeditor.GetObjectsInGroup("Solids")):
            self._object_type = "Solid"

But I use ironpython to print out the list of oEditor.GetObjectsInGroup("Solids") and found that it contains "plastic", which confuses me, am I understanding the process correctly?
 I am looking forward to your answer, thank you!

**Reply by lorenzovecchietti** [reply]
Firstly, are you sure that you want to automatically fix intersections? The code will automatically assign intersections based on the order in which the object appear in the log coming from the intersection file. 

> Then, in the add_priority function, the object named "plastic" reports the error 'NoneType' object has no attribute 'object_type' 

This suggests that `self._app.modeler[new_obj_list[0]]` returns `None`. Can you check what `new_obj_list[0]` is in debug mode and if it is actually present in the pyaedt modeler class?
